### PR TITLE
Add workflow for autodeploying documentation to github pages via an action

### PR DESCRIPTION
**Pull request by morganlee123**
Add workflow for autodeploying documentation to github pages via an action

## Description

Added file .github/workflows/documentation-build.yml and compiled some other actions to properly build and deploy the sphinx documentation

## Motivation and Context

Removes the need for Microsoft Azure to deploy our github pages. (which requires github secret tokens and is all around a mess) 

## How Has This Been Tested?

It has been thoroughly tested on my fork with passing tests and proper deployment to morganlee123.github.io/tardis

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have assigned/requested two reviewers for this pull request.

### CODE REVIEW

File: .github/workflows/documentation-build.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+      - name: Build Sphinx Documentation
+        shell: bash -l {0}
+        run: python setup.py build_docs
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: docs/_build/html/
```
Comment (epassaro) [inline .github/workflows/documentation-build.yml:32]: Use `publish_branch: gh-pages` explicitly as described [here](https://github.com/marketplace/actions/github-pages-action#%EF%B8%8F-set-another-github-pages-branch-publish_branch). It's the default behavior, but it's clearer for people looking for the first time this pipeline.

File: .github/workflows/documentation-build.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  documentation_build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup TARDIS Environment 
+        uses: conda-incubator/setup-miniconda@v1 
+        with:
+            environment-file: tardis_env3.yml
```
Comment (epassaro) [inline .github/workflows/documentation-build.yml:17]: Try adding the `mamba-version` parameter as described [here](https://github.com/conda-incubator/setup-miniconda/#example-6-mamba). I think it's not working now, but just try.

**Comment by epassaro** [comment]
Update the documentation [here](http://tardis-sn.github.io/tardis/development/continuous_integration.html#documentation-pipeline) to reflect this is already done, it's just a line.

**Comment by morganlee123** [comment]
Changes completed. Ready for merge to prod

**Review by epassaro — APPROVED** [review]

**Pull request merged**